// zmp-vue/esm/components/accordion-item.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [_renderSlot(_ctx.$slots, "default")], 2);
}

import { ref, computed, onBeforeUnmount, onMounted } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
import { zmp, zmpready } from '../shared/zmp';
export default {
  name: 'zmp-accordion-item',
  render: render,
  props: _extends({
    opened: Boolean
  }, colorProps),
  emits: ['accordion:beforeopen', 'accordion:open', 'accordion:opened', 'accordion:beforeclose', 'accordion:close', 'accordion:closed'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var elRef = ref(null);

    var onBeforeOpen = function onBeforeOpen(el, prevent) {
      if (elRef.value !== el) return;
      emit('accordion:beforeopen', prevent);
    };

    var onOpen = function onOpen(el) {
      if (elRef.value !== el) return;
      emit('accordion:open');
    };

    var onOpened = function onOpened(el) {
      if (elRef.value !== el) return;
      emit('accordion:opened');
    };

    var onBeforeClose = function onBeforeClose(el, prevent) {
      if (elRef.value !== el) return;
      emit('accordion:beforeclose', prevent);
    };

    var onClose = function onClose(el) {
      if (elRef.value !== el) return;
      emit('accordion:close');
    };

    var onClosed = function onClosed(el) {
      if (elRef.value !== el) return;
      emit('accordion:closed');
    };

    var attachEvents = function attachEvents() {
      zmpready(function () {
        zmp.on('accordionBeforeOpen', onBeforeOpen);
        zmp.on('accordionOpen', onOpen);
        zmp.on('accordionOpened', onOpened);
        zmp.on('accordionBeforeClose', onBeforeClose);
        zmp.on('accordionClose', onClose);
        zmp.on('accordionClosed', onClosed);
      });
    };

    var detachEvents = function detachEvents() {
      zmp.off('accordionBeforeOpen', onBeforeOpen);
      zmp.off('accordionOpen', onOpen);
      zmp.off('accordionOpened', onOpened);
      zmp.off('accordionBeforeClose', onBeforeClose);
      zmp.off('accordionClose', onClose);
      zmp.off('accordionClosed', onClosed);
    };

    onMounted(function () {
      return attachEvents();
    });
    onBeforeUnmount(function () {
      return detachEvents();
    });
    var classes = computed(function () {
      return classNames('accordion-item', {
        'accordion-item-opened': props.opened
      }, colorClasses(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
