// zmp-vue/esm/components/menu-dropdown.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("div", {
    class: _ctx.classes
  }, [_createVNode("div", {
    class: "menu-dropdown-content",
    style: {
      height: _ctx.contentHeight
    }
  }, [_renderSlot(_ctx.$slots, "default")], 4)], 2);
}

import { computed } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
export default {
  name: 'zmp-menu-dropdown',
  render: render,
  props: _extends({
    contentHeight: String,
    position: String,
    left: Boolean,
    center: Boolean,
    right: Boolean
  }, colorProps),
  setup: function setup(props) {
    var classes = computed(function () {
      var positionComputed = props.position || 'left';
      if (props.left) positionComputed = 'left';
      if (props.center) positionComputed = 'center';
      if (props.right) positionComputed = 'right';
      return classNames('menu-dropdown', "menu-dropdown-" + positionComputed, colorClasses(props));
    });
    return {
      classes: classes
    };
  }
};
