// zmp-vue/esm/components/messages.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [_renderSlot(_ctx.$slots, "default")], 2);
}

import { computed, ref, onMounted, onBeforeUnmount, onBeforeUpdate, onUpdated, watch } from 'vue';
import { classNames, noUndefinedProps } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
import { zmpready, zmp } from '../shared/zmp';
export default {
  name: 'zmp-messages',
  render: render,
  props: _extends({
    autoLayout: {
      type: Boolean,
      default: false
    },
    messages: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    newMessagesFirst: {
      type: Boolean,
      default: false
    },
    scrollMessages: {
      type: Boolean,
      default: true
    },
    scrollMessagesOnEdge: {
      type: Boolean,
      default: true
    },
    typing: {
      type: Boolean,
      default: false
    },
    firstMessageRule: Function,
    lastMessageRule: Function,
    tailMessageRule: Function,
    sameNameMessageRule: Function,
    sameHeaderMessageRule: Function,
    sameFooterMessageRule: Function,
    sameAvatarMessageRule: Function,
    customClassMessageRule: Function,
    renderMessage: Function,
    init: {
      type: Boolean,
      default: true
    }
  }, colorProps),
  setup: function setup(props) {
    var zmpMessages = null;
    var childrenBeforeUpdated = null;
    var elRef = ref(null);
    onMounted(function () {
      if (!props.init) return;
      zmpready(function () {
        zmpMessages = zmp.messages.create(noUndefinedProps({
          el: elRef.value,
          autoLayout: props.autoLayout,
          messages: props.messages,
          newMessagesFirst: props.newMessagesFirst,
          scrollMessages: props.scrollMessages,
          scrollMessagesOnEdge: props.scrollMessagesOnEdge,
          firstMessageRule: props.firstMessageRule,
          lastMessageRule: props.lastMessageRule,
          tailMessageRule: props.tailMessageRule,
          sameNameMessageRule: props.sameNameMessageRule,
          sameHeaderMessageRule: props.sameHeaderMessageRule,
          sameFooterMessageRule: props.sameFooterMessageRule,
          sameAvatarMessageRule: props.sameAvatarMessageRule,
          customClassMessageRule: props.customClassMessageRule,
          renderMessage: props.renderMessage
        }));

        if (zmpMessages && props.typing) {
          zmpMessages.showTyping();
        }
      });
    });
    onBeforeUpdate(function () {
      if (!props.init || !elRef.value) return;
      var children = elRef.value.children;
      if (!children) return;
      childrenBeforeUpdated = children.length;

      for (var i = 0; i < children.length; i += 1) {
        children[i].classList.add('message-appeared');
      }
    });
    onUpdated(function () {
      if (!props.init) return;
      if (!elRef.value) return;
      var children = elRef.value.children;
      if (!children) return;
      var childerAftterUpdated = children.length;

      for (var i = 0; i < children.length; i += 1) {
        if (!children[i].classList.contains('message-appeared')) {
          children[i].classList.add('message-appear-from-bottom');
        }
      }

      if (zmpMessages && zmpMessages.layout && props.autoLayout) {
        zmpMessages.layout();
      }

      if (childerAftterUpdated !== childrenBeforeUpdated && zmpMessages && zmpMessages.scroll && props.scrollMessages) {
        zmpMessages.scroll();
      }
    });
    onBeforeUnmount(function () {
      if (zmpMessages && zmpMessages.destroy) zmpMessages.destroy();
      zmpMessages = null;
    });
    watch(function () {
      return props.typing;
    }, function (newValue) {
      if (!zmpMessages) return;
      if (newValue) zmpMessages.showTyping();else zmpMessages.hideTyping();
    });
    var classes = computed(function () {
      return classNames('messages', colorClasses(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
