// zmp-vue/esm/components/messagebar-sheet-image.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { createVNode as _createVNode, renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

var _hoisted_1 = /*#__PURE__*/_createVNode("i", {
  class: "icon icon-checkbox"
}, null, -1);

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("label", {
    class: _ctx.classes,
    style: _ctx.styles
  }, [_createVNode("input", {
    type: "checkbox",
    checked: _ctx.checked,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["checked"]), _hoisted_1, _renderSlot(_ctx.$slots, "default")], 6);
}

import { computed } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
export default {
  name: 'zmp-messagebar-sheet-image',
  render: render,
  props: _extends({
    image: String,
    checked: Boolean
  }, colorProps),
  emits: ['checked', 'unchecked', 'change', 'update:checked'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;

    var onChange = function onChange(event) {
      if (event.target.checked) emit('checked', event);else emit('unchecked', event);
      emit('update:checked', event.target.checked);
      emit('change', event);
    };

    var classes = computed(function () {
      return classNames('messagebar-sheet-image', 'checkbox', colorClasses(props));
    });
    var styles = computed(function () {
      return {
        backgroundImage: props.image && "url(" + props.image + ")"
      };
    });
    return {
      classes: classes,
      styles: styles,
      onChange: onChange
    };
  }
};
