// zmp-vue/esm/components/popover.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { createVNode as _createVNode, renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

var _hoisted_1 = /*#__PURE__*/_createVNode("div", {
  class: "popover-angle"
}, null, -1);

var _hoisted_2 = {
  class: "popover-inner"
};

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [_hoisted_1, _createVNode("div", _hoisted_2, [_renderSlot(_ctx.$slots, "default")])], 2);
}

import { computed, ref, watch, onMounted, onBeforeUnmount } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
import { zmpready, zmp } from '../shared/zmp';
import { modalStateClasses } from '../shared/modal-state-classes';
export default {
  name: 'zmp-popover',
  render: render,
  props: _extends({
    opened: Boolean,
    animate: {
      type: Boolean,
      default: undefined
    },
    targetEl: {
      type: [String, Object],
      default: undefined
    },
    backdrop: {
      type: Boolean,
      default: undefined
    },
    backdropEl: {
      type: [String, Object],
      default: undefined
    },
    closeByBackdropClick: {
      type: Boolean,
      default: undefined
    },
    closeByOutsideClick: {
      type: Boolean,
      default: undefined
    },
    closeOnEscape: {
      type: Boolean,
      default: undefined
    },
    containerEl: {
      type: [String, Object],
      default: undefined
    }
  }, colorProps),
  emits: ['popover:open', 'popover:opened', 'popover:close', 'popover:closed', 'update:opened'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var zmpPopover = ref(null); // eslint-disable-next-line

    var isOpened = props.opened;
    var isClosing = false;
    var elRef = ref(null);

    var onOpen = function onOpen(instance) {
      isOpened = true;
      isClosing = false;
      emit('popover:open', instance);
      emit('update:opened', true);
    };

    var onOpened = function onOpened(instance) {
      emit('popover:opened', instance);
    };

    var onClose = function onClose(instance) {
      isOpened = false;
      isClosing = true;
      emit('popover:close', instance);
    };

    var onClosed = function onClosed(instance) {
      isClosing = false;
      emit('popover:closed', instance);
      emit('update:opened', false);
    };

    watch(function () {
      return props.opened;
    }, function (value) {
      if (!zmpPopover.value) return;

      if (value) {
        zmpPopover.value.open();
      } else {
        zmpPopover.value.close();
      }
    });
    onMounted(function () {
      if (!elRef.value) return;
      var popoverParams = {
        el: elRef.value,
        on: {
          open: onOpen,
          opened: onOpened,
          close: onClose,
          closed: onClosed
        }
      };
      var targetEl = props.targetEl,
          closeByBackdropClick = props.closeByBackdropClick,
          closeByOutsideClick = props.closeByOutsideClick,
          closeOnEscape = props.closeOnEscape,
          backdrop = props.backdrop,
          backdropEl = props.backdropEl,
          containerEl = props.containerEl;
      if (typeof targetEl !== 'undefined') popoverParams.targetEl = targetEl;
      if (typeof closeByBackdropClick !== 'undefined') popoverParams.closeByBackdropClick = closeByBackdropClick;
      if (typeof closeByOutsideClick !== 'undefined') popoverParams.closeByOutsideClick = closeByOutsideClick;
      if (typeof closeOnEscape !== 'undefined') popoverParams.closeOnEscape = closeOnEscape;
      if (typeof backdrop !== 'undefined') popoverParams.backdrop = backdrop;
      if (typeof backdropEl !== 'undefined') popoverParams.backdropEl = backdropEl;
      if (typeof containerEl !== 'undefined') popoverParams.containerEl = containerEl;
      zmpready(function () {
        zmpPopover.value = zmp.popover.create(popoverParams);

        if (props.opened) {
          zmpPopover.value.open(targetEl, false);
        }
      });
    });
    onBeforeUnmount(function () {
      if (zmpPopover.value) {
        zmpPopover.value.destroy();
      }

      zmpPopover.value = null;
    });
    var classes = computed(function () {
      return classNames('popover', modalStateClasses({
        isOpened: isOpened,
        isClosing: isClosing
      }), colorClasses(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
