// zmp-vue/esm/components/radio.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { createVNode as _createVNode, renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

var _hoisted_1 = /*#__PURE__*/_createVNode("i", {
  class: "icon-radio"
}, null, -1);

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("label", {
    class: _ctx.classes
  }, [_createVNode("input", {
    ref: "inputElRef",
    type: "radio",
    name: _ctx.name,
    value: _ctx.value,
    disabled: _ctx.disabled,
    readonly: _ctx.readonly,
    checked: _ctx.checked,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["name", "value", "disabled", "readonly", "checked"]), _hoisted_1, _renderSlot(_ctx.$slots, "default")], 2);
}

import { ref, computed } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
export default {
  name: 'zmp-radio',
  render: render,
  props: _extends({
    checked: Boolean,
    name: [Number, String],
    value: {
      type: [Number, String, Boolean],
      default: undefined
    },
    disabled: Boolean,
    readonly: Boolean
  }, colorProps),
  emits: ['update:checked', 'change'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var inputElRef = ref(null);

    var onChange = function onChange(event) {
      emit('update:checked', event.target.checked);
      emit('change', event);
    };

    var classes = computed(function () {
      return classNames({
        radio: true,
        disabled: props.disabled
      }, colorClasses(props));
    });
    return {
      inputElRef: inputElRef,
      classes: classes,
      onChange: onChange
    };
  }
};
