// zmp-vue/esm/components/tabs.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock, createCommentVNode as _createCommentVNode } from "vue";

function render(_ctx, _cache) {
  return _ctx.animated || _ctx.swipeable ? (_openBlock(), _createBlock("div", {
    key: 0,
    ref: "elRef",
    class: _ctx.classNames(_ctx.wrapClasses, _ctx.classes)
  }, [_createVNode("div", {
    class: _ctx.tabsClasses
  }, [_renderSlot(_ctx.$slots, "default")], 2)], 2)) : (_openBlock(), _createBlock("div", {
    key: 1,
    ref: "elRef",
    class: _ctx.classNames(_ctx.tabsClasses, _ctx.classes)
  }, [_renderSlot(_ctx.$slots, "default")], 2));
}

import { computed, ref, onMounted } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
export default {
  name: 'zmp-tabs',
  render: render,
  props: _extends({
    animated: Boolean,
    swipeable: Boolean,
    routable: Boolean,
    swiperParams: {
      type: Object,
      default: undefined
    }
  }, colorProps),
  setup: function setup(props) {
    var elRef = ref(null);
    onMounted(function () {
      if (!props.swipeable || !props.swiperParams) return;
      if (!elRef.value) return;
      elRef.value.zmpSwiperParams = props.swiperParams;
    });
    var classes = computed(function () {
      return classNames(colorClasses(props));
    });
    var wrapClasses = computed(function () {
      return classNames({
        'tabs-animated-wrap': props.animated,
        'tabs-swipeable-wrap': props.swipeable
      });
    });
    var tabsClasses = computed(function () {
      return classNames({
        tabs: true,
        'tabs-routable': props.routable
      });
    });
    return {
      elRef: elRef,
      classes: classes,
      wrapClasses: wrapClasses,
      tabsClasses: tabsClasses,
      classNames: classNames
    };
  }
};
