// zmp-vue/esm/components/toggle.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from "vue";

var _hoisted_1 = /*#__PURE__*/_createVNode("span", {
  class: "toggle-icon"
}, null, -1);

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("label", {
    ref: "elRef",
    class: _ctx.classes
  }, [_createVNode("input", {
    type: "checkbox",
    name: _ctx.name,
    disabled: _ctx.disabled,
    readonly: _ctx.readonly,
    checked: _ctx.checked,
    value: _ctx.value,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["name", "disabled", "readonly", "checked", "value"]), _hoisted_1], 2);
}

import { computed, ref, watch, onMounted, onBeforeUnmount } from 'vue';
import { classNames } from '../shared/utils';
import { colorClasses, colorProps } from '../shared/mixins';
import { zmpready, zmp } from '../shared/zmp';
import { useTooltip } from '../shared/use-tooltip';
export default {
  name: 'zmp-toggle',
  render: render,
  props: _extends({
    init: {
      type: Boolean,
      default: true
    },
    checked: Boolean,
    disabled: Boolean,
    readonly: Boolean,
    name: String,
    value: [String, Number, Array],
    tooltip: String,
    tooltipTrigger: String
  }, colorProps),
  emits: ['change', 'toggle:change', 'update:checked'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var zmpToggle = null;
    var elRef = ref(null);
    useTooltip(elRef, props);

    var onChange = function onChange(event) {
      emit('change', event);
    };

    watch(function () {
      return props.checked;
    }, function (newValue) {
      if (!zmpToggle) return;
      zmpToggle.checked = newValue;
    });
    onMounted(function () {
      zmpready(function () {
        if (!props.init || !elRef.value) return;
        zmpToggle = zmp.toggle.create({
          el: elRef.value,
          on: {
            change: function change(toggleInstance) {
              emit('toggle:change', toggleInstance.checked);
              emit('update:checked', toggleInstance.checked);
            }
          }
        });
      });
    });
    onBeforeUnmount(function () {
      if (zmpToggle && zmpToggle.destroy && zmpToggle.$el) {
        zmpToggle.destroy();
      }

      zmpToggle = null;
    });
    var classes = computed(function () {
      return classNames('toggle', {
        disabled: props.disabled
      }, colorClasses(props));
    });
    return {
      classes: classes,
      elRef: elRef,
      zmpToggle: zmpToggle,
      onChange: onChange
    };
  }
};
