// zmp-vue/esm/zalo-ui-kits/app.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, resolveComponent as _resolveComponent, withCtx as _withCtx, renderList as _renderList, createSlots as _createSlots, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  var _component_App = _resolveComponent("App");

  return _openBlock(), _createBlock(_component_App, _ctx.zmpParams, _createSlots({
    _: 2
  }, [_renderList(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: _withCtx(function (scope) {
        return [_renderSlot(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040);
}

import App from '../components/app';
import routerConfig from '../shared/router-config';
export default {
  name: 'zmp-app',
  render: render,
  components: {
    App: App
  },
  props: App.props,
  emits: App.emits,
  setup: function setup(props) {
    var navbar = props.navbar,
        routes = props.routes;

    var navbarConfigs = _extends({}, navbar, {
      iosCenterTitle: navbar && navbar.iosCenterTitle || false
    });

    var appRoutes = routerConfig() || routes;
    return {
      zmpParams: _extends({}, props, {
        routes: appRoutes,
        navbar: navbarConfigs,
        iosTranslucentBars: false,
        iosTranslucentModals: false
      })
    };
  }
};
