// zmp-vue/esm/zalo-ui-kits/checkbox.js
import { toDisplayString as _toDisplayString, openBlock as _openBlock, createBlock as _createBlock, createCommentVNode as _createCommentVNode, renderSlot as _renderSlot, resolveComponent as _resolveComponent, mergeProps as _mergeProps, withCtx as _withCtx } from "vue";
var _hoisted_1 = {
  key: 0,
  class: "zmp-checkbox-label"
};

function render(_ctx, _cache) {
  var _component_Checkbox = _resolveComponent("Checkbox");

  return _openBlock(), _createBlock(_component_Checkbox, _mergeProps(_ctx.$attrs, {
    class: "no-ripple"
  }), {
    default: _withCtx(function () {
      return [_ctx.label ? (_openBlock(), _createBlock("span", _hoisted_1, _toDisplayString(_ctx.label), 1)) : _createCommentVNode("", true), _renderSlot(_ctx.$slots, "default")];
    }),
    _: 3
  }, 16);
}

import Checkbox from '../components/checkbox';
export default {
  name: 'zmp-checkbox',
  render: render,
  components: {
    Checkbox: Checkbox
  },
  props: {
    label: String
  },
  setup: function setup(props) {}
};
