// zmp-vue/esm/zalo-ui-kits/list.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, resolveComponent as _resolveComponent, mergeProps as _mergeProps, withCtx as _withCtx, renderList as _renderList, createSlots as _createSlots, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  var _class;

  var _component_List = _resolveComponent("List");

  return _openBlock(), _createBlock(_component_List, _mergeProps(_extends({}, _ctx.$attrs, _ctx.$props), {
    class: (_class = {
      'skeleton-text': _ctx.loading
    }, _class["skeleton-effect-" + _ctx.skeletonEffect] = _ctx.loading && _ctx.isValidEffectName, _class)
  }), _createSlots({
    _: 2
  }, [_renderList(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: _withCtx(function (scope) {
        return [_renderSlot(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040, ["class"]);
}

import { computed } from 'vue';
import List from '../components/list';
import { validateSkeletonEffect } from '../../common/utils';
export default {
  name: 'zmp-list',
  render: render,
  components: {
    List: List
  },
  props: {
    loading: Boolean,
    skeletonEffect: String
  },
  setup: function setup(props) {
    var isValidEffectName = computed(function () {
      return validateSkeletonEffect(props.skeletonEffect);
    });
    return {
      isValidEffectName: isValidEffectName
    };
  }
};
