// zmp-vue/esm/zalo-ui-kits/row.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, resolveComponent as _resolveComponent, mergeProps as _mergeProps, withCtx as _withCtx, renderList as _renderList, createSlots as _createSlots, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  var _class;

  var _component_Row = _resolveComponent("Row");

  return _openBlock(), _createBlock(_component_Row, _mergeProps({
    class: (_class = {}, _class["flex-gap-" + _ctx.gap] = !_ctx.noGap && _ctx.gap, _class)
  }, _extends({}, _ctx.$props, _ctx.$attrs)), _createSlots({
    _: 2
  }, [_renderList(_ctx.$slots, function (_, name) {
    return {
      name: name,
      fn: _withCtx(function (slotData) {
        return [_renderSlot(_ctx.$slots, name, slotData)];
      })
    };
  })]), 1040, ["class"]);
}

import Row from '../components/row';
export default {
  name: 'zmp-row',
  render: render,
  props: {
    gap: [String, Number],
    noGap: Boolean
  },
  components: {
    Row: Row
  },
  setup: function setup(props) {}
};
