// zmp-vue/esm/zalo-ui-kits/searchbar.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { renderSlot as _renderSlot, resolveComponent as _resolveComponent, mergeProps as _mergeProps, withCtx as _withCtx, renderList as _renderList, createSlots as _createSlots, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  var _component_SearchBar = _resolveComponent("SearchBar");

  return _openBlock(), _createBlock(_component_SearchBar, _mergeProps(_extends({}, _ctx.$attrs, _ctx.$props), {
    class: {
      'zmp-searchbar-small': _ctx.small,
      disabled: _ctx.disabled
    },
    disableButton: false,
    noHairline: ""
  }), _createSlots({
    _: 2
  }, [_renderList(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: _withCtx(function (scope) {
        return [_renderSlot(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040, ["class"]);
}

import SearchBar from '../components/searchbar';
export default {
  name: 'zmp-searchbar',
  render: render,
  components: {
    SearchBar: SearchBar
  },
  props: {
    small: Boolean,
    disabled: Boolean
  },
  setup: function setup(props) {}
};
