// zmp-vue/esm/zalo-ui-kits/sheet.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { resolveComponent as _resolveComponent, createVNode as _createVNode, withCtx as _withCtx, openBlock as _openBlock, createBlock as _createBlock, createCommentVNode as _createCommentVNode, toDisplayString as _toDisplayString, renderSlot as _renderSlot, renderList as _renderList, createSlots as _createSlots } from "vue";
var _hoisted_1 = {
  key: 0,
  class: "zmp-sheet-default-toolbar"
};
var _hoisted_2 = {
  key: 1,
  class: "zmp-sheet-default-title-container"
};
var _hoisted_3 = {
  class: "zmp-sheet-default-title"
};
var _hoisted_4 = {
  key: 0,
  class: "zmp-sheet-default-subtitle"
};

function render(_ctx, _cache) {
  var _component_zmp_icon = _resolveComponent("zmp-icon");

  var _component_zmp_link = _resolveComponent("zmp-link");

  var _component_Sheet = _resolveComponent("Sheet");

  return _openBlock(), _createBlock(_component_Sheet, _extends({}, _ctx.$props, _ctx.$attrs), _createSlots({
    fixed: _withCtx(function () {
      return [_ctx.title || _ctx.closeButton ? (_openBlock(), _createBlock("div", _hoisted_1, [_ctx.closeButton ? (_openBlock(), _createBlock(_component_zmp_link, {
        key: 0,
        sheetClose: "",
        class: "zmp-sheet-default-close-button no-ripple"
      }, {
        default: _withCtx(function () {
          return [_createVNode(_component_zmp_icon, {
            zmp: "zi-close"
          })];
        }),
        _: 1
      })) : _createCommentVNode("", true), _ctx.title ? (_openBlock(), _createBlock("div", _hoisted_2, [_createVNode("div", _hoisted_3, _toDisplayString(_ctx.title), 1), _ctx.subtitle ? (_openBlock(), _createBlock("div", _hoisted_4, _toDisplayString(_ctx.subtitle), 1)) : _createCommentVNode("", true)])) : _createCommentVNode("", true)])) : _createCommentVNode("", true)];
    }),
    _: 2
  }, [_renderList(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: _withCtx(function (scope) {
        return [_renderSlot(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040);
}

import Sheet from '../components/sheet';
export default {
  name: 'zmp-sheet',
  render: render,
  components: {
    Sheet: Sheet
  },
  props: _extends({}, Sheet.props, {
    title: String,
    subtitle: String,
    closeButton: Boolean
  })
};
