// zmp-vue/esm/zalo-ui-kits/tabbar.js
import { renderSlot as _renderSlot, resolveComponent as _resolveComponent, mergeProps as _mergeProps, withCtx as _withCtx, renderList as _renderList, createSlots as _createSlots, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  var _component_Toolbar = _resolveComponent("Toolbar");

  return _openBlock(), _createBlock(_component_Toolbar, _mergeProps({
    "onToolbar:hide": _cache[1] || (_cache[1] = function ($event) {
      return _ctx.$emit('tabbar:hide');
    }),
    "onToolbar:show": _cache[2] || (_cache[2] = function ($event) {
      return _ctx.$emit('tabbar:show');
    }),
    tabbar: "",
    class: {
      'toolbar-fill': _ctx.fill
    }
  }, _ctx.$attrs), _createSlots({
    _: 2
  }, [_renderList(_ctx.$slots, function (_, name) {
    return {
      name: name,
      fn: _withCtx(function (slotData) {
        return [_renderSlot(_ctx.$slots, name, slotData)];
      })
    };
  })]), 1040, ["class"]);
}

import Toolbar from '../components/toolbar';
export default {
  name: 'zmp-tabbar',
  render: render,
  components: {
    Toolbar: Toolbar
  },
  props: {
    fill: Boolean
  },
  emits: ['tabbar:hide', 'tabbar:show'],
  setup: function setup() {}
};
