// zmp-vue/esm/zalo-ui-kits/title.js
import { renderSlot as _renderSlot, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock("div", {
    class: _ctx.classes
  }, [_renderSlot(_ctx.$slots, "default")], 2);
}

import { computed } from 'vue';
import { classNames } from '../shared/utils';
export default {
  name: 'zmp-box',
  render: render,
  props: {
    size: String,
    bold: Boolean,
    fontSize: [String, Number],
    noSpace: Boolean
  },
  setup: function setup(props) {
    var classes = computed(function () {
      return classNames('typo-heading', {
        'typo-heading-bold': props.bold,
        'typo-heading-xsmall': props.size === 'xsmall',
        'typo-heading-small': props.size === 'small',
        'typo-heading-normal': !props.size || props.size === 'normal',
        'typo-heading-large': props.size === 'large',
        'typo-heading-xlarge': props.size === 'xlarge',
        'typo-heading-center': props.textAlign === 'center',
        'typo-heading-left': props.textAlign === 'left',
        'typo-heading-right': props.textAlign === 'right',
        'typo-heading-no-spacing': props.noSpacing
      });
    });
    return {
      classes: classes
    };
  }
};
