// zmp-vue/cjs/shared/plugin.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _lite = _interopRequireDefault(require("zmp-core/lite"));

var _componentsRouter = _interopRequireDefault(require("./components-router"));

var _zmp = require("./zmp");

exports.zmp = _zmp.zmp;
exports.zmpready = _zmp.zmpready;
exports.theme = _zmp.theme;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ZMPVue = {
  name: 'vuePlugin',
  installed: false,
  install: function install(params) {
    if (params === void 0) {
      params = {};
    }

    if (ZMPVue.installed) return;
    ZMPVue.installed = true;
    (0, _zmp.zmpinitEvents)();
    var _params = params,
        paramsTheme = _params.theme,
        userAgent = _params.userAgent;
    if (paramsTheme === 'md') _zmp.theme.md = true;
    if (paramsTheme === 'ios') _zmp.theme.ios = true;
    if (paramsTheme === 'aurora') _zmp.theme.aurora = true; // eslint-disable-next-line

    var needThemeCalc = typeof window === 'undefined' ? !!userAgent : true;

    if (needThemeCalc && (!paramsTheme || paramsTheme === 'auto')) {
      var device = _lite.default.getDevice({
        userAgent: userAgent
      }, true);

      _zmp.theme.ios = !!device.ios;
      _zmp.theme.aurora = device.desktop && device.electron;
      _zmp.theme.md = !_zmp.theme.ios && !_zmp.theme.aurora;
    }

    (0, _zmp.zmpready)(function () {
      (0, _zmp.setTheme)();
    });

    _lite.default.Router.use(_componentsRouter.default);
  }
};
var _default = ZMPVue;
exports.default = _default;
