// zmp-vue/cjs/zalo-ui-kits/actions-button.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _actionsButton = _interopRequireDefault(require("../components/actions-button"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _component_ActionsButton = (0, _vue.resolveComponent)("ActionsButton");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_ActionsButton, (0, _vue.mergeProps)(_extends({}, _ctx.$props, _ctx.$attrs), {
    class: {
      'negative-button': _ctx.negative
    }
  }), (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: (0, _vue.withCtx)(function (scope) {
        return [(0, _vue.renderSlot)(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040, ["class"]);
}

var _default = {
  name: 'zmp-actions-button',
  render: render,
  components: {
    ActionsButton: _actionsButton.default
  },
  props: _extends({}, _actionsButton.default.props, {
    negative: Boolean
  })
};
exports.default = _default;
