// zmp-vue/cjs/zalo-ui-kits/app.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _app = _interopRequireDefault(require("../components/app"));

var _routerConfig = _interopRequireDefault(require("../shared/router-config"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _component_App = (0, _vue.resolveComponent)("App");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_App, _ctx.zmpParams, (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: (0, _vue.withCtx)(function (scope) {
        return [(0, _vue.renderSlot)(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040);
}

var _default = {
  name: 'zmp-app',
  render: render,
  components: {
    App: _app.default
  },
  props: _app.default.props,
  emits: _app.default.emits,
  setup: function setup(props) {
    var navbar = props.navbar,
        routes = props.routes;

    var navbarConfigs = _extends({}, navbar, {
      iosCenterTitle: navbar && navbar.iosCenterTitle || false
    });

    var appRoutes = (0, _routerConfig.default)() || routes;
    return {
      zmpParams: _extends({}, props, {
        routes: appRoutes,
        navbar: navbarConfigs,
        iosTranslucentBars: false,
        iosTranslucentModals: false
      })
    };
  }
};
exports.default = _default;
