// zmp-vue/cjs/zalo-ui-kits/card.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _hoisted_1 = {
  key: 0,
  class: "zmp-card-title"
};
var _hoisted_2 = {
  class: "zmp-card-body"
};

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes,
    style: _ctx.style
  }, [_ctx.title || _ctx.$slots.title ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_1, [_ctx.$slots.title ? (0, _vue.renderSlot)(_ctx.$slots, "title", {
    key: 0
  }) : ((0, _vue.openBlock)(), (0, _vue.createBlock)(_vue.Fragment, {
    key: 1
  }, [(0, _vue.createTextVNode)((0, _vue.toDisplayString)(_ctx.title), 1)], 64))])) : (0, _vue.createCommentVNode)("", true), (0, _vue.createVNode)("div", _hoisted_2, [(0, _vue.renderSlot)(_ctx.$slots, "default")])], 6);
}

var _default = {
  name: 'zmp-card',
  render: render,
  props: {
    inset: Boolean,
    title: String
  },
  setup: function setup(props) {
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('zmp-card', {
        'zmp-card-inset': props.inset
      });
    });
    return {
      classes: classes
    };
  }
};
exports.default = _default;
