// zmp-vue/cjs/zalo-ui-kits/checkbox.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _checkbox = _interopRequireDefault(require("../components/checkbox"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _hoisted_1 = {
  key: 0,
  class: "zmp-checkbox-label"
};

function render(_ctx, _cache) {
  var _component_Checkbox = (0, _vue.resolveComponent)("Checkbox");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Checkbox, (0, _vue.mergeProps)(_ctx.$attrs, {
    class: "no-ripple"
  }), {
    default: (0, _vue.withCtx)(function () {
      return [_ctx.label ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_1, (0, _vue.toDisplayString)(_ctx.label), 1)) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "default")];
    }),
    _: 3
  }, 16);
}

var _default = {
  name: 'zmp-checkbox',
  render: render,
  components: {
    Checkbox: _checkbox.default
  },
  props: {
    label: String
  },
  setup: function setup(props) {}
};
exports.default = _default;
