// zmp-vue/cjs/zalo-ui-kits/grid-item.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _link = _interopRequireDefault(require("./link"));

var _icon = _interopRequireDefault(require("./icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  class: "grid-item-icon"
};
var _hoisted_2 = {
  key: 1,
  class: "grid-item-label"
};

function render(_ctx, _cache) {
  var _component_Icon = (0, _vue.resolveComponent)("Icon");

  var _component_Link = (0, _vue.resolveComponent)("Link");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Link, (0, _vue.mergeProps)({
    class: "grid-item no-ripple",
    "no-link-class": "",
    href: _ctx.to || '#'
  }, _extends({}, _ctx.$props, _ctx.$attrs)), {
    default: (0, _vue.withCtx)(function () {
      return [_ctx.icon || _ctx.$slots.icon ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_1, [_ctx.$slots.icon ? (0, _vue.renderSlot)(_ctx.$slots, "icon", {
        key: 0
      }) : ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Icon, {
        key: 1,
        zmp: _ctx.icon
      }, null, 8, ["zmp"]))])) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "default"), _ctx.label ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_2, (0, _vue.toDisplayString)(_ctx.label), 1)) : (0, _vue.createCommentVNode)("", true)];
    }),
    _: 3
  }, 16, ["href"]);
}

var _default = {
  name: 'zmp-grid-item',
  render: render,
  components: {
    Link: _link.default,
    Icon: _icon.default
  },
  props: {
    label: String,
    icon: String,
    to: String,
    key: String,
    disabled: Boolean
  },
  setup: function setup() {}
};
exports.default = _default;
