// zmp-vue/cjs/zalo-ui-kits/link.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _zmp = require("../shared/zmp");

var _utils = require("../shared/utils");

var _utils2 = require("../../common/utils");

var _mixins = require("../shared/mixins");

var _useLink2 = require("../shared/use-link");

var _useIcon = require("../shared/use-icon");

var _useRouteProps = require("../shared/use-route-props");

var _useTooltip = require("../shared/use-tooltip");

var _useSmartSelect = require("../shared/use-smart-select");

var _badge = _interopRequireDefault(require("../components/badge"));

var _useIcon2 = _interopRequireDefault(require("../components/use-icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _component_zmp_use_icon = (0, _vue.resolveComponent)("zmp-use-icon");

  var _component_zmp_badge = (0, _vue.resolveComponent)("zmp-badge");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)("a", (0, _vue.mergeProps)({
    ref: "elRef",
    class: _ctx.classes
  }, _ctx.attrs, {
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
    })
  }), [_ctx.icon ? ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_zmp_use_icon, {
    key: 0,
    icon: _ctx.icon
  }, null, 8, ["icon"])) : (0, _vue.createCommentVNode)("", true), _ctx.text ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", {
    key: 1,
    class: _ctx.isTabbarLabel ? 'tabbar-label' : ''
  }, [(0, _vue.createTextVNode)((0, _vue.toDisplayString)(_ctx.text) + " ", 1), _ctx.badge ? ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_zmp_badge, {
    key: 0,
    color: _ctx.badgeColor
  }, {
    default: (0, _vue.withCtx)(function () {
      return [(0, _vue.createTextVNode)((0, _vue.toDisplayString)(_ctx.badge), 1)];
    }),
    _: 1
  }, 8, ["color"])) : (0, _vue.createCommentVNode)("", true)], 2)) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "default")], 16);
}

var _default = {
  name: 'zmp-link',
  render: render,
  components: {
    zmpBadge: _badge.default,
    zmpUseIcon: _useIcon2.default
  },
  props: _extends({
    noLinkClass: Boolean,
    text: String,
    tabLink: [Boolean, String],
    tabLinkActive: Boolean,
    tabbarLabel: Boolean,
    iconOnly: Boolean,
    iconBadge: [String, Number],
    badge: [String, Number],
    badgeColor: [String],
    href: {
      type: [String, Boolean],
      default: '#'
    },
    target: String,
    tooltip: String,
    tooltipTrigger: String,
    smartSelect: Boolean,
    smartSelectParams: Object
  }, _mixins.iconProps, _mixins.colorProps, _mixins.actionsProps, _mixins.routerProps),
  emits: ['click'],
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        emit = _ref.emit;
    var elRef = (0, _vue.ref)(null);
    var zmpSmartSelect = null;
    (0, _useTooltip.useTooltip)(elRef, props);
    (0, _useRouteProps.useRouteProps)(elRef, props);
    (0, _useSmartSelect.useSmartSelect)(props, function (instance) {
      zmpSmartSelect = instance;
    }, function () {
      return elRef.value;
    });
    var TabbarContext = (0, _vue.inject)('TabbarContext', {
      value: {}
    });
    var isTabbarLabel = (0, _vue.computed)(function () {
      return props.tabbarLabel || TabbarContext.value.tabbarHasLabels;
    });
    var defaultHref = (0, _vue.computed)(function () {
      var _zmp$params$clicks$de, _zmp$params$clicks;

      return (_zmp$params$clicks$de = (_zmp$params$clicks = _zmp.zmp.params.clicks) == null ? void 0 : _zmp$params$clicks.defaultHref) != null ? _zmp$params$clicks$de : '#';
    });
    var attrs = (0, _vue.computed)(function () {
      var tabLink = props.tabLink,
          target = props.target;
      return _extends({
        href: defaultHref.value,
        target: target,
        'data-tab': (0, _utils.isStringProp)(tabLink) && tabLink || undefined
      }, (0, _mixins.routerAttrs)(props), (0, _mixins.actionsAttrs)(props));
    });
    var paths = (0, _vue.computed)(function () {
      var _zmp$routes, _zmp$views$routes, _zmp$views;

      return ((_zmp$routes = _zmp.zmp.routes) != null ? _zmp$routes : []).concat((_zmp$views$routes = (_zmp$views = _zmp.zmp.views[_zmp.zmp.views.length - 1]) == null ? void 0 : _zmp$views.routes) != null ? _zmp$views$routes : []).map(function (route) {
        return route.path;
      });
    });
    var disabledLink = (0, _vue.computed)(function () {
      if (props.href && props.href !== '#' && !(0, _utils2.validateUrl)(props.href, paths.value)) {
        return true;
      }

      return false;
    });
    var classes = (0, _vue.computed)(function () {
      var iconOnly = props.iconOnly,
          text = props.text,
          noLinkClass = props.noLinkClass,
          tabLink = props.tabLink,
          tabLinkActive = props.tabLinkActive,
          smartSelect = props.smartSelect;
      var iconOnlyComputed;
      var hasChildren = slots && slots.default;

      if (iconOnly || !text && !hasChildren) {
        iconOnlyComputed = true;
      } else {
        iconOnlyComputed = false;
      }

      return (0, _utils.classNames)({
        link: !(noLinkClass || isTabbarLabel.value),
        'icon-only': iconOnlyComputed,
        'tab-link': tabLink || tabLink === '',
        'tab-link-active': tabLinkActive,
        'smart-select': smartSelect,
        disabled: disabledLink.value
      }, (0, _mixins.colorClasses)(props), (0, _mixins.routerClasses)(props), (0, _mixins.actionsClasses)(props));
    });
    var icon = (0, _vue.computed)(function () {
      return (0, _useIcon.useIcon)(props);
    });

    var _useLink = (0, _useLink2.useLink)(props),
        linkClick = _useLink[0];

    var handleClick = function handleClick(e) {
      if (props.href && props.href !== '#') {
        linkClick(e);
      }

      emit('click', e);
    };

    return {
      elRef: elRef,
      icon: icon,
      isTabbarLabel: isTabbarLabel,
      attrs: attrs,
      classes: classes,
      zmpSmartSelect: zmpSmartSelect,
      handleClick: handleClick
    };
  }
};
exports.default = _default;
