// zmp-vue/cjs/zalo-ui-kits/list.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _list = _interopRequireDefault(require("../components/list"));

var _utils = require("../../common/utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _class;

  var _component_List = (0, _vue.resolveComponent)("List");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_List, (0, _vue.mergeProps)(_extends({}, _ctx.$attrs, _ctx.$props), {
    class: (_class = {
      'skeleton-text': _ctx.loading
    }, _class["skeleton-effect-" + _ctx.skeletonEffect] = _ctx.loading && _ctx.isValidEffectName, _class)
  }), (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: (0, _vue.withCtx)(function (scope) {
        return [(0, _vue.renderSlot)(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040, ["class"]);
}

var _default = {
  name: 'zmp-list',
  render: render,
  components: {
    List: _list.default
  },
  props: {
    loading: Boolean,
    skeletonEffect: String
  },
  setup: function setup(props) {
    var isValidEffectName = (0, _vue.computed)(function () {
      return (0, _utils.validateSkeletonEffect)(props.skeletonEffect);
    });
    return {
      isValidEffectName: isValidEffectName
    };
  }
};
exports.default = _default;
