// zmp-vue/cjs/zalo-ui-kits/row.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _row = _interopRequireDefault(require("../components/row"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _class;

  var _component_Row = (0, _vue.resolveComponent)("Row");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Row, (0, _vue.mergeProps)({
    class: (_class = {}, _class["flex-gap-" + _ctx.gap] = !_ctx.noGap && _ctx.gap, _class)
  }, _extends({}, _ctx.$props, _ctx.$attrs)), (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, name) {
    return {
      name: name,
      fn: (0, _vue.withCtx)(function (slotData) {
        return [(0, _vue.renderSlot)(_ctx.$slots, name, slotData)];
      })
    };
  })]), 1040, ["class"]);
}

var _default = {
  name: 'zmp-row',
  render: render,
  props: {
    gap: [String, Number],
    noGap: Boolean
  },
  components: {
    Row: _row.default
  },
  setup: function setup(props) {}
};
exports.default = _default;
