// zmp-vue/cjs/zalo-ui-kits/searchbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _searchbar = _interopRequireDefault(require("../components/searchbar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  var _component_SearchBar = (0, _vue.resolveComponent)("SearchBar");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_SearchBar, (0, _vue.mergeProps)(_extends({}, _ctx.$attrs, _ctx.$props), {
    class: {
      'zmp-searchbar-small': _ctx.small,
      disabled: _ctx.disabled
    },
    disableButton: false,
    noHairline: ""
  }), (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: (0, _vue.withCtx)(function (scope) {
        return [(0, _vue.renderSlot)(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040, ["class"]);
}

var _default = {
  name: 'zmp-searchbar',
  render: render,
  components: {
    SearchBar: _searchbar.default
  },
  props: {
    small: Boolean,
    disabled: Boolean
  },
  setup: function setup(props) {}
};
exports.default = _default;
