// zmp-vue/cjs/zalo-ui-kits/sheet.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _sheet = _interopRequireDefault(require("../components/sheet"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  class: "zmp-sheet-default-toolbar"
};
var _hoisted_2 = {
  key: 1,
  class: "zmp-sheet-default-title-container"
};
var _hoisted_3 = {
  class: "zmp-sheet-default-title"
};
var _hoisted_4 = {
  key: 0,
  class: "zmp-sheet-default-subtitle"
};

function render(_ctx, _cache) {
  var _component_zmp_icon = (0, _vue.resolveComponent)("zmp-icon");

  var _component_zmp_link = (0, _vue.resolveComponent)("zmp-link");

  var _component_Sheet = (0, _vue.resolveComponent)("Sheet");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Sheet, _extends({}, _ctx.$props, _ctx.$attrs), (0, _vue.createSlots)({
    fixed: (0, _vue.withCtx)(function () {
      return [_ctx.title || _ctx.closeButton ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_1, [_ctx.closeButton ? ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_zmp_link, {
        key: 0,
        sheetClose: "",
        class: "zmp-sheet-default-close-button no-ripple"
      }, {
        default: (0, _vue.withCtx)(function () {
          return [(0, _vue.createVNode)(_component_zmp_icon, {
            zmp: "zi-close"
          })];
        }),
        _: 1
      })) : (0, _vue.createCommentVNode)("", true), _ctx.title ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_2, [(0, _vue.createVNode)("div", _hoisted_3, (0, _vue.toDisplayString)(_ctx.title), 1), _ctx.subtitle ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_4, (0, _vue.toDisplayString)(_ctx.subtitle), 1)) : (0, _vue.createCommentVNode)("", true)])) : (0, _vue.createCommentVNode)("", true)])) : (0, _vue.createCommentVNode)("", true)];
    }),
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, slot) {
    return {
      name: slot,
      fn: (0, _vue.withCtx)(function (scope) {
        return [(0, _vue.renderSlot)(_ctx.$slots, slot, scope)];
      })
    };
  })]), 1040);
}

var _default = {
  name: 'zmp-sheet',
  render: render,
  components: {
    Sheet: _sheet.default
  },
  props: _extends({}, _sheet.default.props, {
    title: String,
    subtitle: String,
    closeButton: Boolean
  })
};
exports.default = _default;
