// zmp-vue/cjs/zalo-ui-kits/tabbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _toolbar = _interopRequireDefault(require("../components/toolbar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function render(_ctx, _cache) {
  var _component_Toolbar = (0, _vue.resolveComponent)("Toolbar");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)(_component_Toolbar, (0, _vue.mergeProps)({
    "onToolbar:hide": _cache[1] || (_cache[1] = function ($event) {
      return _ctx.$emit('tabbar:hide');
    }),
    "onToolbar:show": _cache[2] || (_cache[2] = function ($event) {
      return _ctx.$emit('tabbar:show');
    }),
    tabbar: "",
    class: {
      'toolbar-fill': _ctx.fill
    }
  }, _ctx.$attrs), (0, _vue.createSlots)({
    _: 2
  }, [(0, _vue.renderList)(_ctx.$slots, function (_, name) {
    return {
      name: name,
      fn: (0, _vue.withCtx)(function (slotData) {
        return [(0, _vue.renderSlot)(_ctx.$slots, name, slotData)];
      })
    };
  })]), 1040, ["class"]);
}

var _default = {
  name: 'zmp-tabbar',
  render: render,
  components: {
    Toolbar: _toolbar.default
  },
  props: {
    fill: Boolean
  },
  emits: ['tabbar:hide', 'tabbar:show'],
  setup: function setup() {}
};
exports.default = _default;
