// zmp-vue/cjs/zalo-ui-kits/title.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-box',
  render: render,
  props: {
    size: String,
    bold: Boolean,
    fontSize: [String, Number],
    noSpace: Boolean
  },
  setup: function setup(props) {
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('typo-heading', {
        'typo-heading-bold': props.bold,
        'typo-heading-xsmall': props.size === 'xsmall',
        'typo-heading-small': props.size === 'small',
        'typo-heading-normal': !props.size || props.size === 'normal',
        'typo-heading-large': props.size === 'large',
        'typo-heading-xlarge': props.size === 'xlarge',
        'typo-heading-center': props.textAlign === 'center',
        'typo-heading-left': props.textAlign === 'left',
        'typo-heading-right': props.textAlign === 'right',
        'typo-heading-no-spacing': props.noSpacing
      });
    });
    return {
      classes: classes
    };
  }
};
exports.default = _default;
