// zmp-vue/cjs/zalo-ui-kits/toast-preloader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _preloader = _interopRequireDefault(require("./preloader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _hoisted_1 = {
  key: 0,
  class: "preloader-toast-text"
};

function render(_ctx, _cache) {
  var _component_Preloader = (0, _vue.resolveComponent)("Preloader");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: ["preloader-toast", {
      'preloader-toast-visible': _ctx.visible
    }]
  }, [(0, _vue.createVNode)(_component_Preloader, _ctx.$attrs, null, 16), _ctx.$slots.default ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_1, [(0, _vue.renderSlot)(_ctx.$slots, "default")])) : (0, _vue.createCommentVNode)("", true)], 2);
}

var _default = {
  name: 'zmp-toast-preloader',
  render: render,
  props: {
    visible: {
      type: Boolean,
      default: false
    }
  },
  components: {
    Preloader: _preloader.default
  },
  setup: function setup() {}
};
exports.default = _default;
