// zmp-vue/cjs/components/actions-label.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes,
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
    })
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-actions-label',
  render: render,
  props: _extends({
    bold: Boolean
  }, _mixins.colorProps),
  emits: ['click'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;

    var onClick = function onClick(e) {
      emit('click', e);
    };

    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('actions-label', {
        'actions-button-bold': props.bold
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes,
      onClick: onClick
    };
  }
};
exports.default = _default;
