// zmp-vue/cjs/components/actions-button.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _zmp = require("../shared/zmp");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  class: "actions-button-media"
};
var _hoisted_2 = {
  class: "actions-button-text"
};

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    ref: "elRef",
    class: _ctx.classes,
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
    })
  }, [_ctx.hasMedia ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_1, [(0, _vue.renderSlot)(_ctx.$slots, "media")])) : (0, _vue.createCommentVNode)("", true), (0, _vue.createVNode)("div", _hoisted_2, [(0, _vue.renderSlot)(_ctx.$slots, "default")])], 2);
}

var _default = {
  name: 'zmp-actions-button',
  render: render,
  props: _extends({
    bold: Boolean,
    close: {
      type: Boolean,
      default: true
    }
  }, _mixins.colorProps),
  emits: ['click'],
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        emit = _ref.emit;
    var elRef = (0, _vue.ref)(null);

    var onClick = function onClick(e) {
      if (elRef.value && props.close && _zmp.zmp) {
        _zmp.zmp.actions.close(_zmp.zmp.$(elRef.value).parents('.actions-modal'));
      }

      emit('click', e);
    };

    var hasMedia = (0, _vue.computed)(function () {
      return slots && !!slots.media;
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)({
        'actions-button': true,
        'actions-button-bold': props.bold
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes,
      elRef: elRef,
      hasMedia: hasMedia,
      onClick: onClick
    };
  }
};
exports.default = _default;
