// zmp-vue/cjs/components/actions.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _zmp = require("../shared/zmp");

var _modalStateClasses = require("../shared/modal-state-classes");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-actions',
  render: render,
  props: _extends({
    tabletFullscreen: Boolean,
    opened: Boolean,
    animate: {
      type: Boolean,
      default: undefined
    },
    grid: Boolean,
    target: {
      type: [String, Object],
      default: undefined
    },
    convertToPopover: {
      type: Boolean,
      default: undefined
    },
    forceToPopover: {
      type: Boolean,
      default: undefined
    },
    backdrop: {
      type: Boolean,
      default: undefined
    },
    backdropEl: {
      type: [String, Object],
      default: undefined
    },
    closeByBackdropClick: {
      type: Boolean,
      default: undefined
    },
    closeByOutsideClick: {
      type: Boolean,
      default: undefined
    },
    closeOnEscape: {
      type: Boolean,
      default: undefined
    },
    containerEl: {
      type: [String, Object],
      default: undefined
    }
  }, _mixins.colorProps),
  emits: ['actions:open', 'actions:opened', 'actions:close', 'actions:closed', 'update:opened'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    // eslint-disable-next-line
    var zmpActions = null; // eslint-disable-next-line

    var isOpened = props.opened;
    var isClosing = false;
    var elRef = (0, _vue.ref)(null);

    var onOpen = function onOpen(instance) {
      isOpened = true;
      isClosing = false;
      emit('actions:open', instance);
      emit('update:opened', true);
    };

    var onOpened = function onOpened(instance) {
      emit('actions:opened', instance);
    };

    var onClose = function onClose(instance) {
      isOpened = false;
      isClosing = true;
      emit('actions:close', instance);
    };

    var onClosed = function onClosed(instance) {
      isClosing = false;
      emit('actions:closed', instance);
      emit('update:opened', false);
    };

    (0, _vue.watch)(function () {
      return props.opened;
    }, function (value) {
      if (!zmpActions) return;

      if (value) {
        zmpActions.open();
      } else {
        zmpActions.close();
      }
    });
    (0, _vue.onMounted)(function () {
      if (!elRef.value) return;
      var target = props.target,
          convertToPopover = props.convertToPopover,
          forceToPopover = props.forceToPopover,
          closeByBackdropClick = props.closeByBackdropClick,
          closeByOutsideClick = props.closeByOutsideClick,
          closeOnEscape = props.closeOnEscape,
          backdrop = props.backdrop,
          backdropEl = props.backdropEl,
          grid = props.grid,
          containerEl = props.containerEl;
      var params = {
        el: elRef.value,
        grid: grid,
        on: {
          open: onOpen,
          opened: onOpened,
          close: onClose,
          closed: onClosed
        }
      };
      if (typeof target !== 'undefined') params.target = target;
      if (typeof convertToPopover !== 'undefined') params.convertToPopover = convertToPopover;
      if (typeof forceToPopover !== 'undefined') params.forceToPopover = forceToPopover;
      if (typeof closeByBackdropClick !== 'undefined') params.closeByBackdropClick = closeByBackdropClick;
      if (typeof closeByOutsideClick !== 'undefined') params.closeByOutsideClick = closeByOutsideClick;
      if (typeof closeOnEscape !== 'undefined') params.closeOnEscape = closeOnEscape;
      if (typeof backdrop !== 'undefined') params.backdrop = backdrop;
      if (typeof backdropEl !== 'undefined') params.backdropEl = backdropEl;
      if (typeof containerEl !== 'undefined') params.containerEl = containerEl;
      (0, _zmp.zmpready)(function () {
        zmpActions = _zmp.zmp.popup.create(params);

        if (props.opened) {
          zmpActions.open(false);
        }
      });
    });
    (0, _vue.onBeforeUnmount)(function () {
      if (zmpActions) {
        zmpActions.close();
        zmpActions.destroy();
      }

      zmpActions = null;
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('actions-modal', {
        'actions-grid': props.grid
      }, (0, _modalStateClasses.modalStateClasses)({
        isOpened: isOpened,
        isClosing: isClosing
      }), (0, _mixins.colorClasses)(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
exports.default = _default;
