// zmp-vue/cjs/components/appbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes
  }, [(0, _vue.renderSlot)(_ctx.$slots, "before-inner"), _ctx.inner ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    key: 0,
    class: _ctx.innerClasses
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2)) : (0, _vue.renderSlot)(_ctx.$slots, "default", {
    key: 1
  }), (0, _vue.renderSlot)(_ctx.$slots, "after-inner")], 2);
}

var _default = {
  name: 'zmp-appbar',
  render: render,
  props: _extends({
    noShadow: Boolean,
    noHairline: Boolean,
    inner: {
      type: Boolean,
      default: true
    },
    innerClass: String,
    innerClassName: String
  }, _mixins.colorProps),
  setup: function setup(props) {
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('appbar', {
        'no-shadow': props.noShadow,
        'no-hairline': props.noHairline
      }, (0, _mixins.colorClasses)(props));
    });
    var innerClasses = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('appbar-inner', props.innerClass, props.innerClassName);
    });
    return {
      classes: classes,
      innerClasses: innerClasses
    };
  }
};
exports.default = _default;
