// zmp-vue/cjs/components/fab-button.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _useTooltip = require("../shared/use-tooltip");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  class: "fab-label"
};

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("a", {
    ref: "elRef",
    class: _ctx.classes,
    target: _ctx.target,
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
    })
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default"), _ctx.label ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_1, (0, _vue.toDisplayString)(_ctx.label), 1)) : (0, _vue.createCommentVNode)("", true)], 10, ["target"]);
}

var _default = {
  name: 'zmp-fab-button',
  render: render,
  props: _extends({
    fabClose: Boolean,
    label: String,
    target: String,
    tooltip: String,
    tooltipTrigger: String
  }, _mixins.colorProps),
  emits: ['click'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var elRef = (0, _vue.ref)(null);

    var onClick = function onClick(e) {
      emit('click', e);
    };

    (0, _useTooltip.useTooltip)(elRef, props);
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)({
        'fab-close': props.fabClose,
        'fab-label-button': props.label
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes,
      onClick: onClick,
      elRef: elRef
    };
  }
};
exports.default = _default;
