// zmp-vue/cjs/components/fab.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _useTooltip = require("../shared/use-tooltip");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  name: 'zmp-fab',
  props: _extends({
    morphTo: String,
    href: [Boolean, String],
    target: String,
    text: String,
    position: {
      type: String,
      default: 'right-bottom'
    },
    tooltip: String,
    tooltipTrigger: String
  }, _mixins.colorProps),
  emits: ['click'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit,
        slots = _ref.slots;
    var elRef = (0, _vue.ref)(null);

    var onClick = function onClick(e) {
      emit('click', e);
    };

    (0, _useTooltip.useTooltip)(elRef, props);
    var hrefComputed = (0, _vue.computed)(function () {
      var href = props.href;
      if (href === true) href = '#';
      if (href === false) href = undefined; // no href attribute

      return href;
    });
    return function () {
      var linkChildren = [];
      var rootChildren = [];
      var textEl;
      var linkEl;
      var linkSlots = slots.link,
          defaultSlots = slots.default,
          rootSlots = slots.root,
          textSlots = slots.text;

      if (defaultSlots) {
        defaultSlots().forEach(function (vnode) {
          if (typeof vnode === 'undefined') return;
          var tag = vnode.type && vnode.type.name ? vnode.type.name : vnode.type;
          if (tag === 'FabButtons' || tag === 'zmp-fab-buttons') rootChildren.push(vnode);else linkChildren.push(vnode);
        });
      }

      if (props.text || textSlots) {
        textEl = (0, _vue.h)('div', {
          class: 'fab-text'
        }, [props.text, textSlots && textSlots()]);
      }

      if (linkChildren.length || linkSlots || textEl) {
        linkEl = (0, _vue.h)('a', {
          target: props.target,
          href: hrefComputed.value,
          onClick: onClick
        }, [linkChildren, textEl, linkSlots && linkSlots()]);
      }

      var classes = (0, _utils.classNames)('fab', "fab-" + props.position, {
        'fab-morph': props.morphTo,
        'fab-extended': typeof textEl !== 'undefined'
      }, (0, _mixins.colorClasses)(props));
      return (0, _vue.h)('div', {
        class: classes,
        'data-morph-to': props.morphTo,
        ref: elRef
      }, [linkEl, rootChildren, rootSlots && rootSlots()]);
    };
  }
};
exports.default = _default;
