// zmp-vue/cjs/components/login-screen.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _zmp = require("../shared/zmp");

var _modalStateClasses = require("../shared/modal-state-classes");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-login-screen',
  render: render,
  props: _extends({
    opened: Boolean,
    animate: {
      type: Boolean,
      default: undefined
    },
    containerEl: {
      type: [String, Object],
      default: undefined
    }
  }, _mixins.colorProps),
  emits: ['loginscreen:open', 'loginscreen:opened', 'loginscreen:close', 'loginscreen:closed', 'update:opened'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var zmpLoginScreen = (0, _vue.ref)(null); // eslint-disable-next-line

    var isOpened = props.opened;
    var isClosing = false;
    var elRef = (0, _vue.ref)(null);

    var onOpen = function onOpen(instance) {
      isOpened = true;
      isClosing = false;
      emit('loginscreen:open', instance);
      emit('update:opened', true);
    };

    var onOpened = function onOpened(instance) {
      emit('loginscreen:opened', instance);
    };

    var onClose = function onClose(instance) {
      isOpened = false;
      isClosing = true;
      emit('loginscreen:close', instance);
    };

    var onClosed = function onClosed(instance) {
      isClosing = false;
      emit('loginscreen:closed', instance);
      emit('update:opened', false);
    };

    (0, _vue.watch)(function () {
      return props.opened;
    }, function (value) {
      if (!zmpLoginScreen.value) return;

      if (value) {
        zmpLoginScreen.value.open();
      } else {
        zmpLoginScreen.value.close();
      }
    });
    (0, _vue.onMounted)(function () {
      if (!elRef.value) return;
      (0, _zmp.zmpready)(function () {
        var loginScreenParams = {
          el: elRef.value,
          on: {
            open: onOpen,
            opened: onOpened,
            close: onClose,
            closed: onClosed
          }
        };

        if (typeof props.animate !== 'undefined') {
          loginScreenParams.animate = props.animate;
        }

        if (typeof props.containerEl !== 'undefined') {
          loginScreenParams.containerEl = props.containerEl;
        }

        zmpLoginScreen.value = _zmp.zmp.loginScreen.create(loginScreenParams);

        if (props.opened) {
          zmpLoginScreen.value.open(false);
        }
      });
    });
    (0, _vue.onBeforeUnmount)(function () {
      if (zmpLoginScreen.value) {
        zmpLoginScreen.value.destroy();
      }

      zmpLoginScreen.value = null;
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('login-screen', (0, _modalStateClasses.modalStateClasses)({
        isOpened: isOpened,
        isClosing: isClosing
      }), (0, _mixins.colorClasses)(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
exports.default = _default;
