// zmp-vue/cjs/components/menu-dropdown.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes
  }, [(0, _vue.createVNode)("div", {
    class: "menu-dropdown-content",
    style: {
      height: _ctx.contentHeight
    }
  }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 4)], 2);
}

var _default = {
  name: 'zmp-menu-dropdown',
  render: render,
  props: _extends({
    contentHeight: String,
    position: String,
    left: Boolean,
    center: Boolean,
    right: Boolean
  }, _mixins.colorProps),
  setup: function setup(props) {
    var classes = (0, _vue.computed)(function () {
      var positionComputed = props.position || 'left';
      if (props.left) positionComputed = 'left';
      if (props.center) positionComputed = 'center';
      if (props.right) positionComputed = 'right';
      return (0, _utils.classNames)('menu-dropdown', "menu-dropdown-" + positionComputed, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes
    };
  }
};
exports.default = _default;
