// zmp-vue/cjs/components/messagebar-attachment.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    class: _ctx.classes,
    onClick: _cache[2] || (_cache[2] = function () {
      return _ctx.onClick && _ctx.onClick.apply(_ctx, arguments);
    })
  }, [_ctx.image ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("img", {
    key: 0,
    src: _ctx.image
  }, null, 8, ["src"])) : (0, _vue.createCommentVNode)("", true), _ctx.deletable ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", {
    key: 1,
    class: "messagebar-attachment-delete",
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.onDeleteClick && _ctx.onDeleteClick.apply(_ctx, arguments);
    })
  })) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-messagebar-attachment',
  render: render,
  props: _extends({
    image: String,
    deletable: {
      type: Boolean,
      default: true
    }
  }, _mixins.colorProps),
  emits: ['attachment:click', 'attachment:delete'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;

    var onClick = function onClick(event) {
      emit('attachment:click', event);
    };

    var onDeleteClick = function onDeleteClick(event) {
      emit('attachment:delete', event);
    };

    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('messagebar-attachment', (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes,
      onClick: onClick,
      onDeleteClick: onDeleteClick
    };
  }
};
exports.default = _default;
