// zmp-vue/cjs/components/messagebar-sheet-image.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = /*#__PURE__*/(0, _vue.createVNode)("i", {
  class: "icon icon-checkbox"
}, null, -1);

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("label", {
    class: _ctx.classes,
    style: _ctx.styles
  }, [(0, _vue.createVNode)("input", {
    type: "checkbox",
    checked: _ctx.checked,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["checked"]), _hoisted_1, (0, _vue.renderSlot)(_ctx.$slots, "default")], 6);
}

var _default = {
  name: 'zmp-messagebar-sheet-image',
  render: render,
  props: _extends({
    image: String,
    checked: Boolean
  }, _mixins.colorProps),
  emits: ['checked', 'unchecked', 'change', 'update:checked'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;

    var onChange = function onChange(event) {
      if (event.target.checked) emit('checked', event);else emit('unchecked', event);
      emit('update:checked', event.target.checked);
      emit('change', event);
    };

    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('messagebar-sheet-image', 'checkbox', (0, _mixins.colorClasses)(props));
    });
    var styles = (0, _vue.computed)(function () {
      return {
        backgroundImage: props.image && "url(" + props.image + ")"
      };
    });
    return {
      classes: classes,
      styles: styles,
      onChange: onChange
    };
  }
};
exports.default = _default;
