// zmp-vue/cjs/components/popover.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _zmp = require("../shared/zmp");

var _modalStateClasses = require("../shared/modal-state-classes");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = /*#__PURE__*/(0, _vue.createVNode)("div", {
  class: "popover-angle"
}, null, -1);

var _hoisted_2 = {
  class: "popover-inner"
};

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [_hoisted_1, (0, _vue.createVNode)("div", _hoisted_2, [(0, _vue.renderSlot)(_ctx.$slots, "default")])], 2);
}

var _default = {
  name: 'zmp-popover',
  render: render,
  props: _extends({
    opened: Boolean,
    animate: {
      type: Boolean,
      default: undefined
    },
    targetEl: {
      type: [String, Object],
      default: undefined
    },
    backdrop: {
      type: Boolean,
      default: undefined
    },
    backdropEl: {
      type: [String, Object],
      default: undefined
    },
    closeByBackdropClick: {
      type: Boolean,
      default: undefined
    },
    closeByOutsideClick: {
      type: Boolean,
      default: undefined
    },
    closeOnEscape: {
      type: Boolean,
      default: undefined
    },
    containerEl: {
      type: [String, Object],
      default: undefined
    }
  }, _mixins.colorProps),
  emits: ['popover:open', 'popover:opened', 'popover:close', 'popover:closed', 'update:opened'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var zmpPopover = (0, _vue.ref)(null); // eslint-disable-next-line

    var isOpened = props.opened;
    var isClosing = false;
    var elRef = (0, _vue.ref)(null);

    var onOpen = function onOpen(instance) {
      isOpened = true;
      isClosing = false;
      emit('popover:open', instance);
      emit('update:opened', true);
    };

    var onOpened = function onOpened(instance) {
      emit('popover:opened', instance);
    };

    var onClose = function onClose(instance) {
      isOpened = false;
      isClosing = true;
      emit('popover:close', instance);
    };

    var onClosed = function onClosed(instance) {
      isClosing = false;
      emit('popover:closed', instance);
      emit('update:opened', false);
    };

    (0, _vue.watch)(function () {
      return props.opened;
    }, function (value) {
      if (!zmpPopover.value) return;

      if (value) {
        zmpPopover.value.open();
      } else {
        zmpPopover.value.close();
      }
    });
    (0, _vue.onMounted)(function () {
      if (!elRef.value) return;
      var popoverParams = {
        el: elRef.value,
        on: {
          open: onOpen,
          opened: onOpened,
          close: onClose,
          closed: onClosed
        }
      };
      var targetEl = props.targetEl,
          closeByBackdropClick = props.closeByBackdropClick,
          closeByOutsideClick = props.closeByOutsideClick,
          closeOnEscape = props.closeOnEscape,
          backdrop = props.backdrop,
          backdropEl = props.backdropEl,
          containerEl = props.containerEl;
      if (typeof targetEl !== 'undefined') popoverParams.targetEl = targetEl;
      if (typeof closeByBackdropClick !== 'undefined') popoverParams.closeByBackdropClick = closeByBackdropClick;
      if (typeof closeByOutsideClick !== 'undefined') popoverParams.closeByOutsideClick = closeByOutsideClick;
      if (typeof closeOnEscape !== 'undefined') popoverParams.closeOnEscape = closeOnEscape;
      if (typeof backdrop !== 'undefined') popoverParams.backdrop = backdrop;
      if (typeof backdropEl !== 'undefined') popoverParams.backdropEl = backdropEl;
      if (typeof containerEl !== 'undefined') popoverParams.containerEl = containerEl;
      (0, _zmp.zmpready)(function () {
        zmpPopover.value = _zmp.zmp.popover.create(popoverParams);

        if (props.opened) {
          zmpPopover.value.open(targetEl, false);
        }
      });
    });
    (0, _vue.onBeforeUnmount)(function () {
      if (zmpPopover.value) {
        zmpPopover.value.destroy();
      }

      zmpPopover.value = null;
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('popover', (0, _modalStateClasses.modalStateClasses)({
        isOpened: isOpened,
        isClosing: isClosing
      }), (0, _mixins.colorClasses)(props));
    });
    return {
      elRef: elRef,
      classes: classes
    };
  }
};
exports.default = _default;
