// zmp-vue/cjs/components/radio.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = /*#__PURE__*/(0, _vue.createVNode)("i", {
  class: "icon-radio"
}, null, -1);

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("label", {
    class: _ctx.classes
  }, [(0, _vue.createVNode)("input", {
    ref: "inputElRef",
    type: "radio",
    name: _ctx.name,
    value: _ctx.value,
    disabled: _ctx.disabled,
    readonly: _ctx.readonly,
    checked: _ctx.checked,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["name", "value", "disabled", "readonly", "checked"]), _hoisted_1, (0, _vue.renderSlot)(_ctx.$slots, "default")], 2);
}

var _default = {
  name: 'zmp-radio',
  render: render,
  props: _extends({
    checked: Boolean,
    name: [Number, String],
    value: {
      type: [Number, String, Boolean],
      default: undefined
    },
    disabled: Boolean,
    readonly: Boolean
  }, _mixins.colorProps),
  emits: ['update:checked', 'change'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var inputElRef = (0, _vue.ref)(null);

    var onChange = function onChange(event) {
      emit('update:checked', event.target.checked);
      emit('change', event);
    };

    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)({
        radio: true,
        disabled: props.disabled
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      inputElRef: inputElRef,
      classes: classes,
      onChange: onChange
    };
  }
};
exports.default = _default;
