// zmp-vue/cjs/components/segmented.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  class: "segmented-highlight"
};

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.tag), {
    class: _ctx.classes
  }, {
    default: (0, _vue.withCtx)(function () {
      return [(0, _vue.renderSlot)(_ctx.$slots, "default"), _ctx.strong || _ctx.strongIos || _ctx.strongMd || _ctx.strongAurora ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("span", _hoisted_1)) : (0, _vue.createCommentVNode)("", true)];
    }),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'zmp-segmented',
  render: render,
  props: _extends({
    raised: Boolean,
    raisedIos: Boolean,
    raisedMd: Boolean,
    raisedAurora: Boolean,
    round: Boolean,
    roundIos: Boolean,
    roundMd: Boolean,
    roundAurora: Boolean,
    strong: Boolean,
    strongIos: Boolean,
    strongMd: Boolean,
    strongAurora: Boolean,
    tag: {
      type: String,
      default: 'div'
    }
  }, _mixins.colorProps),
  setup: function setup(props) {
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)({
        segmented: true,
        'segmented-raised': props.raised,
        'segmented-raised-ios': props.raisedIos,
        'segmented-raised-aurora': props.raisedAurora,
        'segmented-raised-md': props.raisedMd,
        'segmented-round': props.round,
        'segmented-round-ios': props.roundIos,
        'segmented-round-aurora': props.roundAurora,
        'segmented-round-md': props.roundMd,
        'segmented-strong': props.strong,
        'segmented-strong-ios': props.strongIos,
        'segmented-strong-md': props.strongMd,
        'segmented-strong-aurora': props.strongAurora
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes
    };
  }
};
exports.default = _default;
