// zmp-vue/cjs/components/toggle.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _zmp = require("../shared/zmp");

var _useTooltip = require("../shared/use-tooltip");

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = /*#__PURE__*/(0, _vue.createVNode)("span", {
  class: "toggle-icon"
}, null, -1);

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)("label", {
    ref: "elRef",
    class: _ctx.classes
  }, [(0, _vue.createVNode)("input", {
    type: "checkbox",
    name: _ctx.name,
    disabled: _ctx.disabled,
    readonly: _ctx.readonly,
    checked: _ctx.checked,
    value: _ctx.value,
    onChange: _cache[1] || (_cache[1] = function () {
      return _ctx.onChange && _ctx.onChange.apply(_ctx, arguments);
    })
  }, null, 40, ["name", "disabled", "readonly", "checked", "value"]), _hoisted_1], 2);
}

var _default = {
  name: 'zmp-toggle',
  render: render,
  props: _extends({
    init: {
      type: Boolean,
      default: true
    },
    checked: Boolean,
    disabled: Boolean,
    readonly: Boolean,
    name: String,
    value: [String, Number, Array],
    tooltip: String,
    tooltipTrigger: String
  }, _mixins.colorProps),
  emits: ['change', 'toggle:change', 'update:checked'],
  setup: function setup(props, _ref) {
    var emit = _ref.emit;
    var zmpToggle = null;
    var elRef = (0, _vue.ref)(null);
    (0, _useTooltip.useTooltip)(elRef, props);

    var onChange = function onChange(event) {
      emit('change', event);
    };

    (0, _vue.watch)(function () {
      return props.checked;
    }, function (newValue) {
      if (!zmpToggle) return;
      zmpToggle.checked = newValue;
    });
    (0, _vue.onMounted)(function () {
      (0, _zmp.zmpready)(function () {
        if (!props.init || !elRef.value) return;
        zmpToggle = _zmp.zmp.toggle.create({
          el: elRef.value,
          on: {
            change: function change(toggleInstance) {
              emit('toggle:change', toggleInstance.checked);
              emit('update:checked', toggleInstance.checked);
            }
          }
        });
      });
    });
    (0, _vue.onBeforeUnmount)(function () {
      if (zmpToggle && zmpToggle.destroy && zmpToggle.$el) {
        zmpToggle.destroy();
      }

      zmpToggle = null;
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('toggle', {
        disabled: props.disabled
      }, (0, _mixins.colorClasses)(props));
    });
    return {
      classes: classes,
      elRef: elRef,
      zmpToggle: zmpToggle,
      onChange: onChange
    };
  }
};
exports.default = _default;
