// zmp-vue/cjs/components/treeview-item.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _utils = require("../shared/utils");

var _mixins = require("../shared/mixins");

var _useIcon = require("../shared/use-icon");

var _zmp = require("../shared/zmp");

var _useIcon2 = _interopRequireDefault(require("./use-icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _hoisted_1 = {
  key: 0,
  className: "treeview-toggle"
};
var _hoisted_2 = {
  className: "treeview-item-content"
};
var _hoisted_3 = {
  className: "treeview-item-label"
};
var _hoisted_4 = {
  key: 0,
  className: "treeview-item-children"
};

function render(_ctx, _cache) {
  var _component_zmp_use_icon = (0, _vue.resolveComponent)("zmp-use-icon");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)("div", {
    ref: "elRef",
    class: _ctx.classes
  }, [((0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.itemRootTag), (0, _vue.mergeProps)({
    class: _ctx.itemRootClasses
  }, _ctx.itemRootAttrs, {
    onClick: _ctx.onClick
  }), {
    default: (0, _vue.withCtx)(function () {
      return [(0, _vue.renderSlot)(_ctx.$slots, "root-start"), _ctx.needToggle ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_1)) : (0, _vue.createCommentVNode)("", true), (0, _vue.createVNode)("div", _hoisted_2, [(0, _vue.renderSlot)(_ctx.$slots, "content-start"), _ctx.icon ? ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_zmp_use_icon, {
        key: 0,
        icon: _ctx.icon
      }, null, 8, ["icon"])) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "media"), (0, _vue.createVNode)("div", _hoisted_3, [(0, _vue.renderSlot)(_ctx.$slots, "label-start"), (0, _vue.createTextVNode)(" " + (0, _vue.toDisplayString)(_ctx.label) + " ", 1), (0, _vue.renderSlot)(_ctx.$slots, "label")]), (0, _vue.renderSlot)(_ctx.$slots, "content"), (0, _vue.renderSlot)(_ctx.$slots, "content-end")]), (0, _vue.renderSlot)(_ctx.$slots, "root"), (0, _vue.renderSlot)(_ctx.$slots, "root-end")];
    }),
    _: 3
  }, 16, ["class", "onClick"])), _ctx.hasChildren ? ((0, _vue.openBlock)(), (0, _vue.createBlock)("div", _hoisted_4, [(0, _vue.renderSlot)(_ctx.$slots, "children-start"), (0, _vue.renderSlot)(_ctx.$slots, "default"), (0, _vue.renderSlot)(_ctx.$slots, "children")])) : (0, _vue.createCommentVNode)("", true)], 2);
}

var _default = {
  name: 'zmp-treeview-item',
  render: render,
  components: {
    zmpUseIcon: _useIcon2.default
  },
  props: _extends({
    toggle: {
      type: Boolean,
      default: undefined
    },
    itemToggle: Boolean,
    selectable: Boolean,
    selected: Boolean,
    opened: Boolean,
    label: String,
    loadChildren: Boolean,
    link: {
      type: [Boolean, String],
      default: undefined
    }
  }, _mixins.colorProps, _mixins.actionsProps, _mixins.iconProps, _mixins.routerProps),
  emits: ['click', 'treeview:open', 'treeview:close', 'treeview:loadchildren'],
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        emit = _ref.emit;
    var elRef = (0, _vue.ref)(null);
    var hasChildren = (0, _vue.computed)(function () {
      return slots.default || slots.children || slots['children-start'];
    });
    var needToggle = (0, _vue.computed)(function () {
      return typeof props.toggle === 'undefined' ? hasChildren.value : props.toggle;
    });
    var icon = (0, _vue.computed)(function () {
      return (0, _useIcon.useIcon)(props);
    });

    var onClick = function onClick(event) {
      emit('click', event);
    };

    var onOpen = function onOpen(el) {
      if (elRef.value !== el) return;
      emit('treeview:open', el);
    };

    var onClose = function onClose(el) {
      if (elRef.value !== el) return;
      emit('treeview:close', el);
    };

    var onLoadChildren = function onLoadChildren(el, done) {
      if (elRef.value !== el) return;
      emit('treeview:loadchildren', el, done);
    };

    var attachEvents = function attachEvents() {
      if (!elRef.value) return;
      (0, _zmp.zmpready)(function () {
        _zmp.zmp.on('treeviewOpen', onOpen);

        _zmp.zmp.on('treeviewClose', onClose);

        _zmp.zmp.on('treeviewLoadChildren', onLoadChildren);
      });
    };

    var detachEvents = function detachEvents() {
      if (!_zmp.zmp) return;

      _zmp.zmp.off('treeviewOpen', onOpen);

      _zmp.zmp.off('treeviewClose', onClose);

      _zmp.zmp.off('treeviewLoadChildren', onLoadChildren);
    };

    (0, _vue.onMounted)(function () {
      return attachEvents();
    });
    (0, _vue.onBeforeUnmount)(function () {
      return detachEvents();
    });
    var classes = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('treeview-item', {
        'treeview-item-opened': props.opened,
        'treeview-load-children': props.loadChildren
      }, (0, _mixins.colorClasses)(props));
    });
    var itemRootClasses = (0, _vue.computed)(function () {
      return (0, _utils.classNames)('treeview-item-root', {
        'treeview-item-selectable': props.selectable,
        'treeview-item-selected': props.selected,
        'treeview-item-toggle': props.itemToggle
      }, (0, _mixins.routerClasses)(props), (0, _mixins.actionsClasses)(props));
    });
    var itemRootTag = (0, _vue.computed)(function () {
      return props.link || props.link === '' ? 'a' : 'div';
    });
    var itemRootAttrs = (0, _vue.computed)(function () {
      var href = props.link;
      if (props.link === true) href = '#';
      if (props.link === false) href = undefined; // no href attribute

      return _extends({
        href: href
      }, (0, _mixins.routerAttrs)(props), (0, _mixins.actionsAttrs)(props));
    });
    return {
      itemRootTag: itemRootTag,
      itemRootAttrs: itemRootAttrs,
      itemRootClasses: itemRootClasses,
      classes: classes,
      icon: icon,
      onClick: onClick,
      hasChildren: hasChildren,
      needToggle: needToggle,
      elRef: elRef
    };
  }
};
exports.default = _default;
