// zmp-vue/esm/zmp-vue-bundle.js
/**
 * ZMP Vue 2.8.9
 * Build full featured iOS & Android apps using ZMP & Vue
 * https://h5.zalo.me/vue/
 *
 * Copyright 2014-2023 Zalo Mini Platform
 *
 * Released under the MIT License
 *
 * Released on: January 4, 2023
 */
import ZMPVue from './shared/plugin';
import { zmp, zmpready, theme } from './shared/zmp';
import { useStore } from './shared/use-store';

import AccordionContent from './zalo-ui-kits/accordion-content';
import AccordionItem from './zalo-ui-kits/accordion-item';
import AccordionToggle from './zalo-ui-kits/accordion-toggle';
import Accordion from './zalo-ui-kits/accordion';
import ActionsButton from './zalo-ui-kits/actions-button';
import ActionsGroup from './zalo-ui-kits/actions-group';
import ActionsLabel from './zalo-ui-kits/actions-label';
import Actions from './zalo-ui-kits/actions';
import App from './zalo-ui-kits/app';
import AvatarGroup from './zalo-ui-kits/avatar-group';
import Avatar from './zalo-ui-kits/avatar';
import Box from './zalo-ui-kits/box';
import Button from './zalo-ui-kits/button';
import Card from './zalo-ui-kits/card';
import Checkbox from './zalo-ui-kits/checkbox';
import Col from './zalo-ui-kits/col';
import DatePicker from './zalo-ui-kits/date-picker';
import FabBackdrop from './zalo-ui-kits/fab-backdrop';
import FabButton from './zalo-ui-kits/fab-button';
import FabButtons from './zalo-ui-kits/fab-buttons';
import Fab from './zalo-ui-kits/fab';
import GridItem from './zalo-ui-kits/grid-item';
import Grid from './zalo-ui-kits/grid';
import Icon from './zalo-ui-kits/icon';
import Input from './zalo-ui-kits/input';
import Link from './zalo-ui-kits/link';
import ListInput from './zalo-ui-kits/list-input';
import ListItem from './zalo-ui-kits/list-item';
import List from './zalo-ui-kits/list';
import NavLeft from './zalo-ui-kits/nav-left';
import NavRight from './zalo-ui-kits/nav-right';
import NavTitleLarge from './zalo-ui-kits/nav-title-large';
import NavTitle from './zalo-ui-kits/nav-title';
import Navbar from './zalo-ui-kits/navbar';
import Page from './zalo-ui-kits/page';
import Picker from './zalo-ui-kits/picker';
import Preloader from './zalo-ui-kits/preloader';
import Progressbar from './zalo-ui-kits/progressbar';
import Radio from './zalo-ui-kits/radio';
import Range from './zalo-ui-kits/range';
import Row from './zalo-ui-kits/row';
import Searchbar from './zalo-ui-kits/searchbar';
import Sheet from './zalo-ui-kits/sheet';
import SkeletonAvatar from './zalo-ui-kits/skeleton-avatar';
import SkeletonBlock from './zalo-ui-kits/skeleton-block';
import SkeletonImage from './zalo-ui-kits/skeleton-image';
import SkeletonText from './zalo-ui-kits/skeleton-text';
import SwiperSlide from './zalo-ui-kits/swiper-slide';
import Swiper from './zalo-ui-kits/swiper';
import Tab from './zalo-ui-kits/tab';
import Tabbar from './zalo-ui-kits/tabbar';
import Tabs from './zalo-ui-kits/tabs';
import Text from './zalo-ui-kits/text';
import Title from './zalo-ui-kits/title';
import ToastPreloader from './zalo-ui-kits/toast-preloader';
import Toggle from './zalo-ui-kits/toggle';
import View from './zalo-ui-kits/view';

export { AccordionContent, AccordionItem, AccordionToggle, Accordion, ActionsButton, ActionsGroup, ActionsLabel, Actions, App, AvatarGroup, Avatar, Box, Button, Card, Checkbox, Col, DatePicker, FabBackdrop, FabButton, FabButtons, Fab, GridItem, Grid, Icon, Input, Link, ListInput, ListItem, List, NavLeft, NavRight, NavTitleLarge, NavTitle, Navbar, Page, Picker, Preloader, Progressbar, Radio, Range, Row, Searchbar, Sheet, SkeletonAvatar, SkeletonBlock, SkeletonImage, SkeletonText, SwiperSlide, Swiper, Tab, Tabbar, Tabs, Text, Title, ToastPreloader, Toggle, View }
export { zmp, zmpready, theme, useStore };
export default ZMPVue;

function registerComponents(app) {
  app.component('zmp-accordion-content', AccordionContent);
  app.component('zmp-accordion-item', AccordionItem);
  app.component('zmp-accordion-toggle', AccordionToggle);
  app.component('zmp-accordion', Accordion);
  app.component('zmp-actions-button', ActionsButton);
  app.component('zmp-actions-group', ActionsGroup);
  app.component('zmp-actions-label', ActionsLabel);
  app.component('zmp-actions', Actions);
  app.component('zmp-app', App);
  app.component('zmp-avatar-group', AvatarGroup);
  app.component('zmp-avatar', Avatar);
  app.component('zmp-box', Box);
  app.component('zmp-button', Button);
  app.component('zmp-card', Card);
  app.component('zmp-checkbox', Checkbox);
  app.component('zmp-col', Col);
  app.component('zmp-date-picker', DatePicker);
  app.component('zmp-fab-backdrop', FabBackdrop);
  app.component('zmp-fab-button', FabButton);
  app.component('zmp-fab-buttons', FabButtons);
  app.component('zmp-fab', Fab);
  app.component('zmp-grid-item', GridItem);
  app.component('zmp-grid', Grid);
  app.component('zmp-icon', Icon);
  app.component('zmp-input', Input);
  app.component('zmp-link', Link);
  app.component('zmp-list-input', ListInput);
  app.component('zmp-list-item', ListItem);
  app.component('zmp-list', List);
  app.component('zmp-nav-left', NavLeft);
  app.component('zmp-nav-right', NavRight);
  app.component('zmp-nav-title-large', NavTitleLarge);
  app.component('zmp-nav-title', NavTitle);
  app.component('zmp-navbar', Navbar);
  app.component('zmp-page', Page);
  app.component('zmp-picker', Picker);
  app.component('zmp-preloader', Preloader);
  app.component('zmp-progressbar', Progressbar);
  app.component('zmp-radio', Radio);
  app.component('zmp-range', Range);
  app.component('zmp-row', Row);
  app.component('zmp-searchbar', Searchbar);
  app.component('zmp-sheet', Sheet);
  app.component('zmp-skeleton-avatar', SkeletonAvatar);
  app.component('zmp-skeleton-block', SkeletonBlock);
  app.component('zmp-skeleton-image', SkeletonImage);
  app.component('zmp-skeleton-text', SkeletonText);
  app.component('zmp-swiper-slide', SwiperSlide);
  app.component('zmp-swiper', Swiper);
  app.component('zmp-tab', Tab);
  app.component('zmp-tabbar', Tabbar);
  app.component('zmp-tabs', Tabs);
  app.component('zmp-text', Text);
  app.component('zmp-title', Title);
  app.component('zmp-toast-preloader', ToastPreloader);
  app.component('zmp-toggle', Toggle);
  app.component('zmp-view', View)
}
export { registerComponents }
