// zmp-vue/cjs/zmp-vue-bundle.js
/**
 * ZMP Vue 2.8.9
 * Build full featured iOS & Android apps using ZMP & Vue
 * https://h5.zalo.me/vue/
 *
 * Copyright 2014-2023 Zalo Mini Platform
 *
 * Released under the MIT License
 *
 * Released on: January 4, 2023
 */
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _plugin = _interopRequireDefault(require("./shared/plugin"));

var _zmp = require("./shared/zmp");

exports.zmp = _zmp.zmp;
exports.zmpready = _zmp.zmpready;
exports.theme = _zmp.theme;

var _useStore = require("./shared/use-store");

exports.useStore = _useStore.useStore;

var _accordionContent = _interopRequireDefault(require("./zalo-ui-kits/accordion-content"));

exports.AccordionContent = _accordionContent.default;

var _accordionItem = _interopRequireDefault(require("./zalo-ui-kits/accordion-item"));

exports.AccordionItem = _accordionItem.default;

var _accordionToggle = _interopRequireDefault(require("./zalo-ui-kits/accordion-toggle"));

exports.AccordionToggle = _accordionToggle.default;

var _accordion = _interopRequireDefault(require("./zalo-ui-kits/accordion"));

exports.Accordion = _accordion.default;

var _actionsButton = _interopRequireDefault(require("./zalo-ui-kits/actions-button"));

exports.ActionsButton = _actionsButton.default;

var _actionsGroup = _interopRequireDefault(require("./zalo-ui-kits/actions-group"));

exports.ActionsGroup = _actionsGroup.default;

var _actionsLabel = _interopRequireDefault(require("./zalo-ui-kits/actions-label"));

exports.ActionsLabel = _actionsLabel.default;

var _actions = _interopRequireDefault(require("./zalo-ui-kits/actions"));

exports.Actions = _actions.default;

var _app = _interopRequireDefault(require("./zalo-ui-kits/app"));

exports.App = _app.default;

var _avatarGroup = _interopRequireDefault(require("./zalo-ui-kits/avatar-group"));

exports.AvatarGroup = _avatarGroup.default;

var _avatar = _interopRequireDefault(require("./zalo-ui-kits/avatar"));

exports.Avatar = _avatar.default;

var _box = _interopRequireDefault(require("./zalo-ui-kits/box"));

exports.Box = _box.default;

var _button = _interopRequireDefault(require("./zalo-ui-kits/button"));

exports.Button = _button.default;

var _card = _interopRequireDefault(require("./zalo-ui-kits/card"));

exports.Card = _card.default;

var _checkbox = _interopRequireDefault(require("./zalo-ui-kits/checkbox"));

exports.Checkbox = _checkbox.default;

var _col = _interopRequireDefault(require("./zalo-ui-kits/col"));

exports.Col = _col.default;

var _datePicker = _interopRequireDefault(require("./zalo-ui-kits/date-picker"));

exports.DatePicker = _datePicker.default;

var _fabBackdrop = _interopRequireDefault(require("./zalo-ui-kits/fab-backdrop"));

exports.FabBackdrop = _fabBackdrop.default;

var _fabButton = _interopRequireDefault(require("./zalo-ui-kits/fab-button"));

exports.FabButton = _fabButton.default;

var _fabButtons = _interopRequireDefault(require("./zalo-ui-kits/fab-buttons"));

exports.FabButtons = _fabButtons.default;

var _fab = _interopRequireDefault(require("./zalo-ui-kits/fab"));

exports.Fab = _fab.default;

var _gridItem = _interopRequireDefault(require("./zalo-ui-kits/grid-item"));

exports.GridItem = _gridItem.default;

var _grid = _interopRequireDefault(require("./zalo-ui-kits/grid"));

exports.Grid = _grid.default;

var _icon = _interopRequireDefault(require("./zalo-ui-kits/icon"));

exports.Icon = _icon.default;

var _input = _interopRequireDefault(require("./zalo-ui-kits/input"));

exports.Input = _input.default;

var _link = _interopRequireDefault(require("./zalo-ui-kits/link"));

exports.Link = _link.default;

var _listInput = _interopRequireDefault(require("./zalo-ui-kits/list-input"));

exports.ListInput = _listInput.default;

var _listItem = _interopRequireDefault(require("./zalo-ui-kits/list-item"));

exports.ListItem = _listItem.default;

var _list = _interopRequireDefault(require("./zalo-ui-kits/list"));

exports.List = _list.default;

var _navLeft = _interopRequireDefault(require("./zalo-ui-kits/nav-left"));

exports.NavLeft = _navLeft.default;

var _navRight = _interopRequireDefault(require("./zalo-ui-kits/nav-right"));

exports.NavRight = _navRight.default;

var _navTitleLarge = _interopRequireDefault(require("./zalo-ui-kits/nav-title-large"));

exports.NavTitleLarge = _navTitleLarge.default;

var _navTitle = _interopRequireDefault(require("./zalo-ui-kits/nav-title"));

exports.NavTitle = _navTitle.default;

var _navbar = _interopRequireDefault(require("./zalo-ui-kits/navbar"));

exports.Navbar = _navbar.default;

var _page = _interopRequireDefault(require("./zalo-ui-kits/page"));

exports.Page = _page.default;

var _picker = _interopRequireDefault(require("./zalo-ui-kits/picker"));

exports.Picker = _picker.default;

var _preloader = _interopRequireDefault(require("./zalo-ui-kits/preloader"));

exports.Preloader = _preloader.default;

var _progressbar = _interopRequireDefault(require("./zalo-ui-kits/progressbar"));

exports.Progressbar = _progressbar.default;

var _radio = _interopRequireDefault(require("./zalo-ui-kits/radio"));

exports.Radio = _radio.default;

var _range = _interopRequireDefault(require("./zalo-ui-kits/range"));

exports.Range = _range.default;

var _row = _interopRequireDefault(require("./zalo-ui-kits/row"));

exports.Row = _row.default;

var _searchbar = _interopRequireDefault(require("./zalo-ui-kits/searchbar"));

exports.Searchbar = _searchbar.default;

var _sheet = _interopRequireDefault(require("./zalo-ui-kits/sheet"));

exports.Sheet = _sheet.default;

var _skeletonAvatar = _interopRequireDefault(require("./zalo-ui-kits/skeleton-avatar"));

exports.SkeletonAvatar = _skeletonAvatar.default;

var _skeletonBlock = _interopRequireDefault(require("./zalo-ui-kits/skeleton-block"));

exports.SkeletonBlock = _skeletonBlock.default;

var _skeletonImage = _interopRequireDefault(require("./zalo-ui-kits/skeleton-image"));

exports.SkeletonImage = _skeletonImage.default;

var _skeletonText = _interopRequireDefault(require("./zalo-ui-kits/skeleton-text"));

exports.SkeletonText = _skeletonText.default;

var _swiperSlide = _interopRequireDefault(require("./zalo-ui-kits/swiper-slide"));

exports.SwiperSlide = _swiperSlide.default;

var _swiper = _interopRequireDefault(require("./zalo-ui-kits/swiper"));

exports.Swiper = _swiper.default;

var _tab = _interopRequireDefault(require("./zalo-ui-kits/tab"));

exports.Tab = _tab.default;

var _tabbar = _interopRequireDefault(require("./zalo-ui-kits/tabbar"));

exports.Tabbar = _tabbar.default;

var _tabs = _interopRequireDefault(require("./zalo-ui-kits/tabs"));

exports.Tabs = _tabs.default;

var _text = _interopRequireDefault(require("./zalo-ui-kits/text"));

exports.Text = _text.default;

var _title = _interopRequireDefault(require("./zalo-ui-kits/title"));

exports.Title = _title.default;

var _toastPreloader = _interopRequireDefault(require("./zalo-ui-kits/toast-preloader"));

exports.ToastPreloader = _toastPreloader.default;

var _toggle = _interopRequireDefault(require("./zalo-ui-kits/toggle"));

exports.Toggle = _toggle.default;

var _view = _interopRequireDefault(require("./zalo-ui-kits/view"));

exports.View = _view.default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = _plugin.default;
exports.default = _default;

exports.registerComponents = function registerComponents (app) {
  app.component('zmp-accordion-content', AccordionContent);
  app.component('zmp-accordion-item', AccordionItem);
  app.component('zmp-accordion-toggle', AccordionToggle);
  app.component('zmp-accordion', Accordion);
  app.component('zmp-actions-button', ActionsButton);
  app.component('zmp-actions-group', ActionsGroup);
  app.component('zmp-actions-label', ActionsLabel);
  app.component('zmp-actions', Actions);
  app.component('zmp-app', App);
  app.component('zmp-avatar-group', AvatarGroup);
  app.component('zmp-avatar', Avatar);
  app.component('zmp-box', Box);
  app.component('zmp-button', Button);
  app.component('zmp-card', Card);
  app.component('zmp-checkbox', Checkbox);
  app.component('zmp-col', Col);
  app.component('zmp-date-picker', DatePicker);
  app.component('zmp-fab-backdrop', FabBackdrop);
  app.component('zmp-fab-button', FabButton);
  app.component('zmp-fab-buttons', FabButtons);
  app.component('zmp-fab', Fab);
  app.component('zmp-grid-item', GridItem);
  app.component('zmp-grid', Grid);
  app.component('zmp-icon', Icon);
  app.component('zmp-input', Input);
  app.component('zmp-link', Link);
  app.component('zmp-list-input', ListInput);
  app.component('zmp-list-item', ListItem);
  app.component('zmp-list', List);
  app.component('zmp-nav-left', NavLeft);
  app.component('zmp-nav-right', NavRight);
  app.component('zmp-nav-title-large', NavTitleLarge);
  app.component('zmp-nav-title', NavTitle);
  app.component('zmp-navbar', Navbar);
  app.component('zmp-page', Page);
  app.component('zmp-picker', Picker);
  app.component('zmp-preloader', Preloader);
  app.component('zmp-progressbar', Progressbar);
  app.component('zmp-radio', Radio);
  app.component('zmp-range', Range);
  app.component('zmp-row', Row);
  app.component('zmp-searchbar', Searchbar);
  app.component('zmp-sheet', Sheet);
  app.component('zmp-skeleton-avatar', SkeletonAvatar);
  app.component('zmp-skeleton-block', SkeletonBlock);
  app.component('zmp-skeleton-image', SkeletonImage);
  app.component('zmp-skeleton-text', SkeletonText);
  app.component('zmp-swiper-slide', SwiperSlide);
  app.component('zmp-swiper', Swiper);
  app.component('zmp-tab', Tab);
  app.component('zmp-tabbar', Tabbar);
  app.component('zmp-tabs', Tabs);
  app.component('zmp-text', Text);
  app.component('zmp-title', Title);
  app.component('zmp-toast-preloader', ToastPreloader);
  app.component('zmp-toggle', Toggle);
  app.component('zmp-view', View)
}
